service/matching/task_writer.go

```diff
@@ -251,11 +251,11 @@ writerLoop:
 			}
 
 			resp, err := w.appendTasks(ctx, tasks)
-			w.sendWriteResponse(reqs, resp, err)
 			// Update the maxReadLevel after the writes are completed.
```

Comment:
Nice, I think this is correct. Please update this comment with something like:
```suggestion
			// Update the maxReadLevel after the writes are completed, but before we send the response,
			// so that taskReader is guaranteed to see the new read level when SpoolTask wakes it up.
```